src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp

```diff
@@ -0,0 +1,183 @@
+  ~ApparentHorizon() = default;
+
+  // clang-tidy non-const reference pointer.
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+  Strahlkorper<Frame> strahlkorper{};
+  ::FastFlow fast_flow{};
+  Verbosity verbosity{Verbosity::Quiet};
+};
+
+template <typename Frame>
+bool operator==(const ApparentHorizon<Frame>& lhs,
+                const ApparentHorizon<Frame>& rhs) noexcept;
+template <typename Frame>
+bool operator!=(const ApparentHorizon<Frame>& lhs,
+                const ApparentHorizon<Frame>& rhs) noexcept;
+
+}  // namespace OptionHolders
+
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \brief Sends points on a trial apparent horizon to an `Interpolator`.
+///
+/// This differs from `KerrHorizon` in the following ways:
+/// - It supplies points on a prolonged Strahlkorper, at a higher resolution
+///   than the Strahlkorper in the DataBox, as needed for horizon finding.
+/// - It uses a `FastFlow` in the DataBox.
+/// - It has different options (including those for `FastFlow`).
+///
+/// Uses:
+/// - DataBox:
+///   - `::Tags::Domain<VolumeDim, Frame>`
+///   - `::Tags::FastFlow`
+///   - `StrahlkorperTags::CartesianCoords<Frame>`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies:
+///   - `Tags::IndicesOfFilledInterpPoints`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// This Action also has an initialize function that adds to the DataBox:
+/// - `StrahlkorperTags::items_tags<Frame>`
+/// - `StrahlkorperTags::compute_items_tags<Frame>`
+/// - `::Tags::FastFlow`
+/// - `::Tags::Verbosity`
+///
+/// For requirements on InterpolationTargetTag, see InterpolationTarget
+template <typename InterpolationTargetTag, typename Frame>
+struct ApparentHorizon {
+  using options_type = OptionHolders::ApparentHorizon<Frame>;
+  using const_global_cache_tags = tmpl::list<InterpolationTargetTag>;
+  using initialization_tags =
+      tmpl::append<StrahlkorperTags::items_tags<Frame>,
+                   tmpl::list<::Tags::FastFlow, ::Tags::Verbosity>,
+                   StrahlkorperTags::compute_items_tags<Frame>>;
+  template <typename DbTags, typename Metavariables>
+  static auto initialize(
+      db::DataBox<DbTags>&& box,
+      const Parallel::ConstGlobalCache<Metavariables>& cache) noexcept {
+    const auto& options = Parallel::get<InterpolationTargetTag>(cache);
+
+    // Put Strahlkorper and its ComputeItems, FastFlow,
+    // and verbosity into a new DataBox.
+    return db::create_from<
+        db::RemoveTags<>,
+        db::AddSimpleTags<tmpl::push_back<StrahlkorperTags::items_tags<Frame>,
+                                          ::Tags::FastFlow, ::Tags::Verbosity>>,
+        db::AddComputeTags<StrahlkorperTags::compute_items_tags<Frame>>>(
+        std::move(box), std::move(options.strahlkorper),
```

Comment:
the items in the const global cache are const, so `std::move` doesn't change them, so the moves aren't necessary. (comment below can also then be removed).

In order to avoid big copies, it's probably best to have a creator class to hold the options and then only make the (possibly large) strahlkorper when it's put into the databox. This last change can easily be in a different PR.